### PR TITLE
fix: "Stuck on Verifying..."

### PR DESCRIPTION
**Pull request by MikaelMayer**
This PR adds no new operational code because the fix is in Boogie:
https://github.com/boogie-org/boogie/pull/491

This fix has been merged, and this PR bumps the Boogie version, so the test included should now pass.

This tests asks the verifier to do something hard, and cancels verification by performing a change on the text.
Until this PR, the verifier is stuck verifying, but with the fix above, it will cancel verification, kill zombie processes, and continue smoothly

### CODE REVIEW

**Review by keyboardDrummer — APPROVED** [review]

File: Source/DafnyLanguageServer.Test/Various/CancelVerification.cs
Comment (camrein) [inline Source/DafnyLanguageServer.Test/Various/CancelVerification.cs:40]: Prefer using `await Task.Delay(5_000);` here

**Review by camrein — CHANGES_REQUESTED** [review]
Only some small inputs, and it appears something's wrong with the tests.
I'm quite impressed that this feature didn't require changes to the language server itself. 😄
This will sure have a huge impact on UX.

**Comment by camrein** [comment]
Just in case you're waiting for the approval: I think you didn't push your latest changes. 😄

**Comment by MikaelMayer** [comment]
Indeed, I did not, because there is still a problem in Boogie where counter-examples are not reported correctly anymore. I'll be waiting on Boogie to be compatible with Dafny again to solve this issue.

File: Source/DafnyLanguageServer/Language/DafnyProgramVerifier.cs
Comment (MikaelMayer) [inline Source/DafnyLanguageServer/Language/DafnyProgramVerifier.cs:102]:
```suggestion
      using (cancellationToken.Register(() => CancelVerification(uniqueRequestId))) {
```

**Review by camrein — APPROVED** [review]

**Pull request merged**
